extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -461,10 +476,12 @@ export class AmpStoryEmbeddedComponent {
     event.preventDefault();
     event.stopPropagation();
 
-    this.setState_(ComponentState.EXPANDED, this.previousTarget_);
+    this.setState_(EmbeddedComponentState.EXPANDED,
+        /** @type {InteractiveComponentDef} */ (this.storeService_.get(
+            StateProperty.INTERACTIVE_COMPONENT_STATE)));
 
-    this.storeService_.dispatch(
-        Action.TOGGLE_EXPANDED_COMPONENT, this.previousTarget_);
+    this.storeService_.dispatch(Action.TOGGLE_INTERACTIVE_COMPONENT,
+        {state: EmbeddedComponentState.EXPANDED});
```

Comment:
I'm a bit confused by this code too: this action is going to trigger `onEmbeddedComponentUpdate_`, but this handler requires `component.element` to be provided.
Should we remove the `this.setState` above, and let the store trigger `onEmbeddedComponentUpdate_` which will perform the expected actions?
It looks like we're trying to do the same thing twice here

Reply:
Changed the logic as chatted offline. Now on every UI change the store will be notified, and will make the appropriate call to `this.setState`.

UI interaction -> store update -> state update